### PR TITLE
[FIX] Use TYPO3 FlashMessageService instead of fakeSchedulerModuleCon…

### LINKED ISSUE #73: Invalid method signature which blocks Scheduler-Creation
Hi,
when I try to add any scheduler task, I'll get this error:

> Fatal error: Declaration of Causal\ImageAutoresize\Task\fakeSchedulerModuleController::addMessage($message, $severity = TYPO3\CMS\Core\Messaging\FlashMessage::OK) must be compatible with TYPO3\CMS\Scheduler\Controller\SchedulerModuleController::addMessage(TYPO3\CMS\Backend\Template\ModuleTemplate $moduleTemplate, string $message, TYPO3\CMS\Core\Type\ContextualFeedbackSeverity $severity = TYPO3\CMS\Core\Type\ContextualFeedbackSeverity::OK): void in /home/___/domains/___/typo3/vendor/causal/image_autoresize/Classes/Task/BatchResizeAdditionalFieldProvider.php on line 168

My setup is TYPO3 12.2 in Composer Mode with PHP 8.1
The extension is in version 2.2.0

### PR DESCRIPTION
**Pull request by tobiasgraeber**
- Fixes #73 - Invalid method signature which blocks Scheduler-Creation
- Also @see: https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/FlashMessages/Index.html

### CODE REVIEW

**Comment by tobiasgraeber** [comment]
Note: This also fixes, that a sheduler task, if on save a issue occures, cannot be saved. For example, if saving a path of a non existend directory in the include/exclude additionalFields of the Task. On TYPO3 v12.x. (At least on current master-branch the case cause issue is not visible, cause fakeScheduler Messages do not work, error is then reported to errorlog)...

File: Classes/Task/BatchResizeAdditionalFieldProvider.php [hunk 1]
```diff
@@ -121,13 +123,21 @@ public function validateAdditionalFields(array &$submittedData, \TYPO3\CMS\Sched
...
+
+                $message = GeneralUtility::makeInstance(
+                    FlashMessage::class,
                     sprintf(
                         $GLOBALS['LANG']->sL('LLL:EXT:image_autoresize/Resources/Private/Language/locallang_mod.xlf:msg.invalidDirectories'),
                         $directory
                     ),
-                    FlashMessage::ERROR
+                    'EXT:image_autoresize: invalidDirectories error',
+                    ContextualFeedbackSeverity::ERROR,
```
Comment (xperseguers) [inline Classes/Task/BatchResizeAdditionalFieldProvider.php:134]: `ContextualFeedbackSeverity` is introduced in TYPO3 v12, thus it breaks compatibility with TYPO3 v10 and TYPO3 v11.

File: Classes/Task/BatchResizeAdditionalFieldProvider.php [hunk 2]
```diff
@@ -136,12 +146,21 @@ public function validateAdditionalFields(array &$submittedData, \TYPO3\CMS\Sched
...
+
+                $message = GeneralUtility::makeInstance(
+                    FlashMessage::class,
                     sprintf(
                         $GLOBALS['LANG']->sL('LLL:EXT:image_autoresize/Resources/Private/Language/locallang_mod.xlf:msg.invalidExcludeDirectories'),
                         $directory
                     ),
-                    FlashMessage::ERROR);
+                    'EXT:image_autoresize: invalidExcludeDirectories error',
+                    ContextualFeedbackSeverity::ERROR,
```
Comment (xperseguers) [inline Classes/Task/BatchResizeAdditionalFieldProvider.php:157]: Same here

**Review by xperseguers — CHANGES_REQUESTED** [review]
Removing the `fakeSchedulerModuleController` looks OK (but will need to be tested back to TYPO3 v10) but introduction of `ContextualFeedbackSeverity` is a no-go as-is. I'd suggest to introduce a `version_compare()` check.

File: Classes/Task/BatchResizeAdditionalFieldProvider.php (hunk 1 above)
Comment (tobiasgraeber) [reply Classes/Task/BatchResizeAdditionalFieldProvider.php:134]: Yes, can be done, i leave that up to you :-)

File: Classes/Task/BatchResizeAdditionalFieldProvider.php (hunk 2 above)
Comment (tobiasgraeber) [reply Classes/Task/BatchResizeAdditionalFieldProvider.php:157]: Yes, can be done, i leave that up to you :-)

**Comment by xperseguers** [comment]
I prefer to go that direction: https://github.com/xperseguers/t3ext-image_autoresize/tree/bugfix/scheduler-80

**Pull request closed without merging**